src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java

```diff
@@ -42,89 +42,133 @@
-          "welcome_message_text", new Parameter()
-                  .setDefaultValue(ParameterValue
-                          .of(String.format("welcome to app %s", timestamp)))
-                  .setConditionalValues(ImmutableMap.<String, ParameterValue>of(
-                          "ios_en",
-                          ParameterValue.of(String.format("welcome to app en %s", timestamp))
-                  ))
-                  .setDescription("text for welcome message!"),
-          "header_text", new Parameter()
-                  .setDefaultValue(ParameterValue.inAppDefault()));
-
-  private static final Map<String, ParameterGroup> PARAMETER_GROUPS = ImmutableMap.of(
-          "new menu", new ParameterGroup()
-                  .setDescription(String.format("New Menu %s", timestamp))
-                  .setParameters(ImmutableMap.of(
-                          "pumpkin_spice_season", new Parameter()
-                                  .setDefaultValue(ParameterValue.of("true"))
-                                  .setDescription("Whether it's currently pumpkin spice season."))
-                  ));
-
   private static final List<Condition> CONDITIONS = ImmutableList.of(
           new Condition("ios_en", "device.os == 'ios' && device.country in ['us', 'uk']")
                   .setTagColor(TagColor.INDIGO),
           new Condition("android_en",
                   "device.os == 'android' && device.country in ['us', 'uk']"));
 
-  private static final Version VERSION = Version
-          .withDescription(String.format("promo config %s", timestamp));
-
   @BeforeClass
   public static void setUpClass() {
     remoteConfig = FirebaseRemoteConfig.getInstance(IntegrationTestUtils.ensureDefaultApp());
   }
 
   @Test
-  public void testTemplateOperations() throws FirebaseRemoteConfigException {
+  public void testValidateTemplate() throws FirebaseRemoteConfigException {
+    final Template inputTemplate = remoteConfig.getTemplate();
+    final Map<String, Parameter> expectedParameters = getParameters();
+    final Map<String, ParameterGroup> expectedParameterGroups = getParameterGroups();
+    final Version expectedVersion = getVersion();
+    inputTemplate.setParameters(expectedParameters)
+            .setParameterGroups(expectedParameterGroups)
+            .setConditions(CONDITIONS)
+            .setVersion(expectedVersion);
+
+    Template validatedTemplate = remoteConfig.validateTemplate(inputTemplate);
+
+    assertEquals(inputTemplate, validatedTemplate);
+  }
+
+  @Test
+  public void testPublishTemplate() throws FirebaseRemoteConfigException {
     // get template to fetch the active template with correct etag
     final Template oldTemplate = remoteConfig.getTemplate();
     final Template inputTemplate = Template.fromJSON(oldTemplate.toJSON());
-    final String versionNumber = oldTemplate.getVersion().getVersionNumber();
-
+    final Map<String, Parameter> parameters = getParameters();
```

Comment:
Probably doesn't need to clone the template

Reply:
Good catch! Thanks!